Src/Support/Google.Apis.Core/Util/HttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+Copyright 2024 Google Inc
+
+Licensed under the Apache License, Version 2.0(the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using System.Net.Http;
+
+namespace Google.Apis.Util;
+
+/// <summary>
+/// Extension methods for <see cref="HttpRequestMessage"/>.
+/// </summary>
+public static class HttpRequestMessageExtensions
```

Comment:
Rather than making this public, I wonder whether it would be better to have the same code in both projects. That way we can remove it entirely (eventually - it may be a very long time) without affecting the public API at all. We only really want it to be *used* in our projects, right?

Reply:
Actually removing it would then be a binary breaking change, so I'd be nervous about doing that in a minor version.

Given that they need to be kept in sync, how about doing this in one source file that's *linked* in all three projects? I don't mind it being in the same namespace in all three assemblies.